infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/configurator/CredentialsSecretConfigurator.java

```diff
@@ -30,27 +36,54 @@
 public class CredentialsSecretConfigurator implements NamespaceConfigurator {
 
   private final CheServerKubernetesClientFactory cheServerKubernetesClientFactory;
+  private final PersonalAccessTokenManager personalAccessTokenManager;
+
+  private static final Map<String, String> SEARCH_LABELS =
+      ImmutableMap.of(
+          "app.kubernetes.io/part-of", "che.eclipse.org",
+          "app.kubernetes.io/component", "scm-personal-access-token");
+  private static final String ANNOTATION_SCM_URL = "che.eclipse.org/scm-url";
+  private static final String MERGED_GIT_CREDENTIALS_SECRET_NAME =
+      "devworkspace-merged-git-credentials";
 
   @Inject
   public CredentialsSecretConfigurator(
-      CheServerKubernetesClientFactory cheServerKubernetesClientFactory) {
+      CheServerKubernetesClientFactory cheServerKubernetesClientFactory,
+      PersonalAccessTokenManager personalAccessTokenManager) {
     this.cheServerKubernetesClientFactory = cheServerKubernetesClientFactory;
+    this.personalAccessTokenManager = personalAccessTokenManager;
   }
 
   @Override
   public void configure(NamespaceResolutionContext namespaceResolutionContext, String namespaceName)
       throws InfrastructureException {
     var client = cheServerKubernetesClientFactory.create();
-    if (client.secrets().inNamespace(namespaceName).withName(CREDENTIALS_SECRET_NAME).get()
-        == null) {
-      Secret secret =
-          new SecretBuilder()
-              .withType("opaque")
-              .withNewMetadata()
-              .withName(CREDENTIALS_SECRET_NAME)
-              .endMetadata()
-              .build();
-      client.secrets().inNamespace(namespaceName).create(secret);
-    }
+    Optional<Secret> mergedCredentialsSecret =
+        client.secrets().inNamespace(namespaceName).list().getItems().stream()
+            .filter(s -> s.getMetadata().getName().equals(MERGED_GIT_CREDENTIALS_SECRET_NAME))
+            .findAny();
```

Comment:
It's been a while since I've used the java API, but isn't it possible to try to directly `get` the secret we're looking for? 
```
client.secrets().inNamespace(namespaceName).withName(MERGED_GIT_CREDENTIALS_SECRET_NAME).get()
```
followed by a check for `null` since it's no longer an Optional

Reply:
removed the leftover